src/devices/Display/Large4Digit14SegmentDisplay.cs

```diff
@@ -0,0 +1,285 @@
+        private static readonly byte[] s_clearBuffer =
+        {
+            0b0000_0000, // Command
+            0b0000_0000, // Data Disp 1
+            0b0000_0000,
+            0b0000_0000, // Data Disp 2
+            0b0000_0000,
+            0b0000_0000, // Data Disp 3
+            0b0000_0000,
+            0b0000_0000, // Data Disp 4
+            0b0000_0000
+        };
+
+        /// <summary>
+        /// Display buffer
+        /// </summary>
+        private readonly byte[] _displayBuffer =
+        {
+            0b0000_0000, // Command
+            0b0000_0000, // Data Disp 1
+            0b0000_0000,
+            0b0000_0000, // Data Disp 2
+            0b0000_0000,
+            0b0000_0000, // Data Disp 3
+            0b0000_0000,
+            0b0000_0000, // Data Disp 4
+            0b0000_0000
+        };
+
+        /// <summary>
+        /// Translate digit number to buffer address
+        /// </summary>
+        /// <param name="digit">digit to translate</param>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="digit"/></exception>
+        private static int TranslateDigitToBufferAddress(int digit) => digit switch
+        {
+            0 => (int)Address.Digit1,
+            1 => (int)Address.Digit2,
+            2 => (int)Address.Digit3,
+            3 => (int)Address.Digit4,
+            _ => throw new ArgumentOutOfRangeException(nameof(digit)),
+        };
+        #endregion
+
+        #region Public members
+
+        /// <summary>
+        /// Initialize 7-Segment display
+        /// </summary>
+        /// <param name="i2cDevice">The <see cref="I2cDevice"/> to create with.</param>
+        public Large4Digit14SegmentDisplay(I2cDevice i2cDevice)
+            : base(i2cDevice)
+        {
+        }
+
+        /// <inheritdoc/>
+        public int NumberOfDigits { get; } = MaxNumberOfDigits;
+
+        /// <summary>
+        /// Gets or sets a single digit's segments by id
+        /// </summary>
+        /// <param name="address">digit address (0..3)</param>
+        /// <returns>Segment in display buffer for the given address</returns>
+        public Segment14 this[int address]
+        {
+            get => (Segment14)_displayBuffer[TranslateDigitToBufferAddress(address)];
+            set
+            {
+                _displayBuffer[TranslateDigitToBufferAddress(address)] = (byte)value;
+                AutoFlush();
+            }
+        }
+
+        /// <inheritdoc/>
+        /// <remarks>Write clears dots, you'll have to reset them afterwards</remarks>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="startAddress"/></exception>
+        /// <exception cref="ArgumentOutOfRangeException"><see cref="Large4Digit7SegmentDisplay"/> only supports <see cref="MaxNumberOfDigits"/> digits</exception>
+        public override void Write(ReadOnlySpan<byte> digits, int startAddress = 0)
+        {
+            Console.WriteLine($"Length: {digits.Length}");
```

Comment:
Please remove any console output from bindings (use a logger if you need to)

Reply:
Whoops....missed this.

Reply:
Done